bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxCommunicator.java

```diff
@@ -0,0 +1,563 @@
+                        out.write(msg);
+                        out.flush();
+                    }
+
+                    // Log message
+                    if (outgoingMessage != null) {
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("->: {}", outgoingMessage.getName());
+                            logger.debug("->: {}", HexUtils
+                                    .bytesToHex(outgoingMessage.getMessageFrameBytes(CaddxProtocol.Binary), " "));
+                        }
+                    }
+                    if (outgoingMessage != null) {
+                        expectedMessageNumbers = outgoingMessage.getReplyMessageNumbers();
+                    }
+                } else {
+                    logger.trace("CaddxCommunicator.run() skipTransmit: true");
+                    skipTransmit = false;
+                }
+
+                // Check for an incoming message
+                CaddxMessage incomingMessage = null;
+                CaddxMessage throwAway = new CaddxMessage(new byte[] { 0x1d }, false);
+                try {
+                    incomingMessage = exchanger.exchange(throwAway, 3, TimeUnit.SECONDS);
+                } catch (TimeoutException e) {
+                    if (expectedMessageNumbers == null) { // Nothing expected, Nothing received we continue
+                        logger.trace("CaddxCommunicator.run(): Nothing expected, Nothing received we continue");
+                        continue;
+                    }
+                    logger.debug("CaddxCommunicator.run() TimeoutException caught.");
+                }
+
+                // Log
+                if (incomingMessage == null) {
+                    logger.debug("CaddxCommunicator.run() NoMessage received.");
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("<-: {}", incomingMessage.getName());
+                        logger.debug("<-: {}",
+                                HexUtils.bytesToHex(incomingMessage.getMessageFrameBytes(CaddxProtocol.Binary), " "));
+                    }
+                }
+
+                // Check if we wait for a reply
+                if (expectedMessageNumbers == null) {
+                    if (incomingMessage != null) { // Nothing expected. Message received.
+                        logger.trace("CaddxCommunicator.run() Nothing expected, Message received");
+
+                        // Check if Acknowledgement handling is required.
+                        if (incomingMessage.hasAcknowledgementFlag()) {
+                            if (incomingMessage.isChecksumCorrect()) {
+                                messages.putFirst(new CaddxMessage(new byte[] { 0x1d }, false)); // send ACK
+                            } else {
+                                messages.putFirst(new CaddxMessage(new byte[] { 0x1d }, false)); // send ACK
+                            }
```

Comment:
Actually i have put it there intentionally. 
I read some posts on the internet that there were cases, the panel would not correctly calculate the checksum. If i would send back NAK instead of ACK the panel would resend the message, effectively falling into an infinite loop.
Having said that, i do not have evidence for this case. I'm more inclined to think that the posts were wrong or that it happened with on older firmware. I have this panel for more than 10 years, so it must have been a very old firmware.

WDYT? Should i return NAK when the checksum is wrong?

Reply:
What I was pointing out is that the code on the if branch and the else branch are identical. There is no point to an if-statement when the same code will be executed regardless.

Reply:
Ok. Changed.